src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -820,11 +820,24 @@ public PolicyResponse retrieveSnapshotPolicies(
               inaccessibleWorkspaces.addAll(wpms.inaccessible());
             });
 
-    return new PolicyResponse()
-        .policies(PolicyUtils.samToTdrPolicyModels(samPolicyModels))
-        .authDomain(authDomain)
-        .workspaces(accessibleWorkspaces)
-        .inaccessibleWorkspaces(inaccessibleWorkspaces);
+    PolicyResponse policyResponse =
+        new PolicyResponse()
+            .policies(PolicyUtils.samToTdrPolicyModels(samPolicyModels))
+            .authDomain(authDomain)
+            .workspaces(accessibleWorkspaces)
+            .inaccessibleWorkspaces(inaccessibleWorkspaces)
+            .inheritedStewards(List.of());
+
+    Dataset sourceDataset = snapshotDao.retrieveSnapshot(snapshotId).getSourceDataset();
+    if (sourceDataset.isInheritSteward()) {
+      iamService.retrievePolicies(userReq, IamResourceType.DATASET, sourceDataset.getId()).stream()
```

Comment:
After manual testing I found that this won't work for all users and I'll need to use the TDR service account here. I'll update the PR with that fix.